gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -341,6 +341,15 @@ def self.run_sig(hook_mod, method_name, original_method, declaration_block)
         Signature.new_untyped(method: original_method)
       end
 
+    current_method = signature.method.owner.instance_method(method_name)
+    file_path, line = current_method.source_location
+
+    if !@was_ever_final_names[method_name] && original_method != current_method &&
+       !file_path.match?(%r{sorbet-runtime[^/]*/lib/})
```

Comment:
The `file_path` for the wrapper method should be exactly equal to `__FILE__`, since this is the same file that creates the wrapper in the first place:
```suggestion
       file_path != __FILE__
```

Reply:
What if `T::Private::Methods` had a constant `WRAPPER_METHOD_SOURCE_FILE = __FILE__` that this referenced as:
```ruby
  file_path != T::Private::Methods::WRAPPER_METHOD_SOURCE_FILE
```
Then you wouldn't have to keep doing regexp checks and still be safe against the code moving.

Reply:
We can do that too. I don't have strong opinions about this one. @froydnj, what do you think?

Reply:
That sounds good to me. Added the comment and the new constant.